lane-layout-gen/src/main.rs

```diff
@@ -0,0 +1,185 @@
+use std::cmp::{min, max};
+use std::collections::HashMap;
+
+use rand::{Rng, prelude::SliceRandom, distributions::Uniform, SeedableRng};
+
+use clap::Parser;
+
+use kdam::par_tqdm;
+
+use rayon::prelude::*;
+
+#[derive(Parser, Debug)]
+struct Args {
+    #[arg(short)]
+    n: u64,
+
+    #[arg(long)]
+    iter: isize,
```

Comment:
should be optional as well. consider if `search_until_d` is set.

Reply:
and `iter` doesn't honor `search_until_d`  
```
➜ ./result/bin/lane-layout-gen -n 2048 --iter 10000 --search-until-d 51
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 10000/10000 [03:23<00:00, 49.22it/s]find a layout with the max distance: 50
```

Reply:
as expected. When search_until_d is set, the search does not stop until finding a small layout